common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -61,4 +63,17 @@ public MetricSet shuffleMetrics() {
     // Return an empty MetricSet by default.
     return () -> Collections.emptyMap();
   }
+
+  /**
+   * Request the local disk directories, which are specified by DiskBlockManager, for the executors
+   * from the external shuffle service (when this is a ExternalBlockStoreClient) or BlockManager
+   * (when this is a NettyBlockTransferService). Note there's only one executor when this is a
+   * NettyBlockTransferService because we ask one specific executor at a time.
```

Comment:
Can you clarify the last sentence here?

Reply:
Good point. It is true that in case of dynamic allocation executors come and go but an executor can be lost without dynamic allocation too and the code is prepared to handle those: it leads to a `FailureFetchResult` so I expect in case of errors we are as good as before. On the other hand this still helps when local block successfully read.

So I cannot see the why dynamic allocation is excluded from this feature.

Reply:
Oh I got this. 

When the external shuffle service is target by this request we can collect the local dirs of multiple executors at once (as all the host local dirs are available in the external shuffle service running on the host as it is central component in this sense on that host).

But here we can request the local dirs for only one executor the one which handles the request itself.

@Ngone51 what about adding an assert here: https://github.com/apache/spark/blob/1126341a680c3a6bea4f792516ea59b54726e11a/core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala#L119
Checking the array only contains one executor ID and it is equal with the `executorId` of the `blockManager`.

Reply:
I added the check to ensure it's the only one executor id but didn't check its equality with blockManager's executor id. Because we only have `BlockDataManager` in `NettyBlockRpcServer` which does not expose executor id. 

I am still wondering whether it's worthwhile to expose it for the sanity check purpose.